### PR TITLE
Heightmaps used for brushes etc. should be normalised to zero.

### LINKED ISSUE #1508: `//br cliff` acts out of shape
### Server Implementation

Spigot

### Server Version

1.18.1

### Describe the bug

brush pushed edges of touched area to infinity

### To Reproduce

equip //br cliff

### Expected behaviour

smooth edge of brushed terrain

### Screenshots / Videos

image

### Error log (if applicable)

_No response_

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/6a8f9b5981c54c9dbb0d275ebcf4f4e2

### Fawe Version

FastAsyncWorldEdit version 2.0.0-SNAPSHOT-44;0994ea1

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit/ and the issue still persists.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dordsor21**
Fixes #1508

### CODE REVIEW

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/math/heightmap/ArrayHeightMap.java [hunk 1]
```diff
@@ -3,27 +3,27 @@
...
     private double rz;
 
     /**
-     * New height map represented by byte array[][] of values x*z to be scaled given a set size
+     * New height map represented by char array[][] of values x*z to be scaled given a set size.
+     * Limited 0->65535
      *
      * @param height array of height values
-     * @param minY   min y value allowed to be set. Inclusive.
-     * @param maxY   max y value allowed to be set. Inclusive.
```
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/math/heightmap/ArrayHeightMap.java:19]: Missing `@param scale`

**Review by NotMyFault — COMMENTED** [review]
LGTM otherwise

**Comment by NotMyFault** [comment]
@SirYwell can you take a look?
Also, conflicted.

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/math/heightmap/ArrayHeightMap.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public void setSize(int size) {
     public double getHeight(int x, int z) {
         x = (int) Math.max(0, Math.min(width - 1, (x + size) * rx));
         z = (int) Math.max(0, Math.min(length - 1, (z + size) * rz));
-        return ((height[x][z] & 0xFF) * size) / 256d;
+        return ((height[x][z] & 0xFFFF) * size) / scale;
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/math/heightmap/ArrayHeightMap.java:41]: the `& 0xFFFF` is not required anymore, as `char` is unsigned already.

**Review by SirYwell — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/math/heightmap/ArrayHeightMap.java (hunk 2 above)
Comment (dordsor21) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/math/heightmap/ArrayHeightMap.java:41]: oop, true

**Review by NotMyFault — APPROVED** [review]

**Review by SirYwell — APPROVED** [review]

**Pull request merged**